src/System.Private.CoreLib/shared/System/Collections/Generic/IDictionary.cs

```diff
@@ -10,7 +11,7 @@ namespace System.Collections.Generic
     // Keys can be any non-null object.  Values can be any object.
     // You can look up a value in an IDictionary via the default indexed
     // property, Items.
-    public interface IDictionary<TKey, TValue> : ICollection<KeyValuePair<TKey, TValue>>
+    public interface IDictionary<TKey, TValue> : ICollection<KeyValuePair<TKey, TValue>> // TODO-NULLABLE-GENERIC: TKey should be non-nullable
```

Comment:
Aren't we using `where TKey : object` for this?

Reply:
Wouldn't that affect people that implement the interface? I don't know, so that's why I ask.

Reply:
> Wouldn't that affect people that implement the interface? I don't know, so that's why I ask.

Yes.  Is that not desired?

Reply:
Isn't that a breaking change? I.e, when someone is already implementing these interfaces, and update to a new version, will it break them even if they don't have nullability context enabled?

While it makes sense since it is documented a key can't be null: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.idictionary-2.add?view=netframework-4.8#exceptions, I just want to make sure that we're OK and document this breaking change if so.

Reply:
> will it break them even if they don't have nullability context enabled?

It shouldn't.

Reply:
I didn't know about it, will be fixed in new iteration